### PR TITLE
Fix a bunch of bugs I found while using my own code ;)

### PR DESCRIPTION
**Pull request by micha7a**
- Fix the rotation in Procrustes (and add tests)
- Add option to the trajectory to not have the constant term
- Change the full rank condition to more readable (and add tests)
- Fix issue with numpy ints versus python ints in the theory (and add tests)
- Fix the problem with the number of measurements in TheoryTesting notebook
- Add (crude) variance plotting to TheoryTesting notebook

### CODE REVIEW

File: TheoryTesting.ipynb [hunk 1]
```diff
@@ -105,7 +106,7 @@
     "        n_measurements=n,\n",
     "        position_wise=False,\n",
     "        n_anchors=n_anchors,\n",
-    "        n_times=params[\"n_times\"],\n",
+    "        n_times=np.Infinity if one_per_time else params[\"n_times\"],\n",
```
Comment (duembgen) [inline TheoryTesting.ipynb:109]: Is there a reason you use np.Infinity and not something more common like "None"?

File: source/constraints.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def get_reduced_constraints(idx_a, idx_f, anchors, frame):
                           axis=1)
 
 
-def get_frame(n_constraints, n_positions):
+def get_frame(n_constraints, n_positions):  # TODO(michalina) allow random times
```
Comment (duembgen) [inline source/constraints.py:247]: Is this here on purpose?

File: source/evaluate_dataset.py [hunk 3]
```diff
@@ -138,10 +139,10 @@ def match_reference(reference, points):
     """
     assert reference.shape == points.shape
     reference_center = np.mean(reference, axis=1)
-    reference -= reference_center[:, None]
+    reference_centered = reference - reference_center[:, None]
     rotation_center = np.mean(points, axis=1)
     points -= rotation_center[:, None]
-    rotation, e = sp.linalg.orthogonal_procrustes(points.T, reference.T)
+    rotation, e = linalg.orthogonal_procrustes(reference_centered.T, points.T)
```
Comment (duembgen) [inline source/evaluate_dataset.py:145]: Was the order here simply wrong?

File: source/constraints.py (hunk 2 above)
Comment (micha7a) [reply source/constraints.py:247]: Ups, sorry :D I mean, I should add random times anyway, but I shouldn't announce it here ;)

File: source/evaluate_dataset.py (hunk 3 above)
Comment (micha7a) [reply source/evaluate_dataset.py:145]: Yep, but if you wanted to just inverse the rotation then the centres were wrong too - something that wouldn't be visible for a centred dataset. I think. At least that's why I think we didn't notice :D

File: source/constraints.py (hunk 2 above)
Comment (micha7a) [inline source/constraints.py:247]:
```suggestion
def get_frame(n_constraints, n_positions):
```

File: TheoryTesting.ipynb (hunk 1 above)
Comment (duembgen) [reply TheoryTesting.ipynb:109]: Ok makes sense!

**Comment by duembgen** [comment]
Ok thanks for answering my questions! Feel free to merge

**Pull request merged**